core/src/main/java/nu/yona/server/subscriptions/service/BuddyInfoChangeMessageDTO.java

```diff
@@ -0,0 +1,139 @@
+import nu.yona.server.messaging.service.MessageService.TheDTOManager;
+import nu.yona.server.messaging.service.SenderInfo;
+import nu.yona.server.subscriptions.entities.BuddyInfoChangeMessage;
+
+@JsonRootName("buddyInfoChangeMessage")
+public class BuddyInfoChangeMessageDTO extends BuddyMessageLinkedUserDTO
+{
+	private static final String PROCESS = "process";
+	private final String newNickname;
+	private final boolean isProcessed;
+
+	private BuddyInfoChangeMessageDTO(UUID id, ZonedDateTime creationTime, boolean isRead, SenderInfo senderInfo, String message,
+			String newNickname, boolean isProcessed)
+	{
+		super(id, creationTime, isRead, senderInfo, message);
+		this.newNickname = newNickname;
+		this.isProcessed = isProcessed;
+	}
+
+	@Override
+	public String getType()
+	{
+		return "BuddyInfoChangeMessage";
+	}
+
+	public String getNewNickname()
+	{
+		return newNickname;
+	}
+
+	@JsonIgnore
+	public boolean isProcessed()
+	{
+		return isProcessed;
+	}
+
+	@Override
+	public Set<String> getPossibleActions()
+	{
+		Set<String> possibleActions = super.getPossibleActions();
+		if (!isProcessed)
+		{
+			possibleActions.add(PROCESS);
+		}
+		return possibleActions;
+	}
+
+	@Override
+	public boolean canBeDeleted()
+	{
+		return this.isProcessed;
+	}
+
+	public static BuddyInfoChangeMessageDTO createInstance(UserDTO actingUser, BuddyInfoChangeMessage messageEntity,
+			SenderInfo senderInfo)
+	{
+		return new BuddyInfoChangeMessageDTO(messageEntity.getID(), messageEntity.getCreationTime(), messageEntity.isRead(),
+				senderInfo, messageEntity.getMessage(), messageEntity.getNewNickname(), messageEntity.isProcessed());
+	}
+
+	@Component
+	private static class Manager extends BuddyMessageDTO.Manager
+	{
+		@Autowired
+		private TheDTOManager theDTOFactory;
+
+		@Autowired
+		private BuddyService buddyService;
+
+		@PostConstruct
+		private void init()
+		{
+			theDTOFactory.addManager(BuddyInfoChangeMessage.class, this);
+		}
+
+		@Override
+		public MessageDTO createInstance(UserDTO actingUser, Message messageEntity)
+		{
+			return BuddyInfoChangeMessageDTO.createInstance(actingUser, (BuddyInfoChangeMessage) messageEntity,
+					getSenderInfo(actingUser, messageEntity));
+		}
+
+		@Override
+		public MessageActionDTO handleAction(UserDTO actingUser, Message messageEntity, String action,
+				MessageActionDTO requestPayload)
+		{
+			switch (action)
+			{
+				case PROCESS:
+					return handleAction_Process(actingUser, (BuddyInfoChangeMessage) messageEntity, requestPayload);
+				default:
+					return super.handleAction(actingUser, messageEntity, action, requestPayload);
+			}
+		}
+
+		private MessageActionDTO handleAction_Process(UserDTO actingUser, BuddyInfoChangeMessage messageEntity,
+				MessageActionDTO requestPayload)
+		{
+			if (messageEntity.getRelatedUserAnonymizedID().isPresent())
+			{
+				buddyService.updateBuddyInfo(actingUser.getID(), messageEntity.getRelatedUserAnonymizedID().get(),
```

Comment:
The intent of this method is broader: to update any information on the `Buddy` entity. Currently it is only nickname, later on it will include photo and maybe other things.